Sources/InteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -681,15 +1098,53 @@ struct UnimplementedService: InteroperabilityTest {
     try await unimplementedServiceClient.unimplementedCall(
       request: ClientRequest.Single(message: Grpc_Testing_Empty())
     ) { response in
-      let result = response.accepted
-      switch result {
-      case .success(_):
-        throw AssertionFailure(
-          message: "The result should be an error."
-        )
+      switch response.accepted {
+      case .success:
+        throw AssertionFailure(message: "The result should be an error.")
       case .failure(let error):
         try assertEqual(error.code, .unimplemented)
       }
     }
   }
 }
+
+/// This test verifies that an RPC request whose lifetime exceeds its configured timeout value
+/// will end with the DeadlineExceeded status.
+///
+/// Server features:
+/// - FullDuplexCall
+///
+/// Procedure:
+/// 1. Client calls FullDuplexCall with the following request and sets its timeout to 1ms
+///    ```
+///    {
+///        payload:{
+///            body: 27182 bytes of zeros
+///        }
+///    }
+///    ```
+/// 2. Client waits
+///
+/// Client asserts:
+/// - Call completed with status DEADLINE_EXCEEDED.
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+class TimeoutOnSleepingServer: InteroperabilityTest {
+  func run(client: GRPCClient) async throws {
+    let testServiceClient = Grpc_Testing_TestService.Client(client: client)
+
+    var callOptions = CallOptions.defaults
+    callOptions.timeout = .nanoseconds(1)
+
+    try await testServiceClient.fullDuplexCall(
+      request: ClientRequest.Stream { _ in },
+      options: callOptions
+    ) { response in
+      switch response.accepted {
+      case .success:
+        throw AssertionFailure(message: "The result should be an error.")
+      case .failure(let error):
+        try assertEqual(error.code, .deadlineExceeded)
```

Comment:
Much like cancellation we can't rely on the client producing this error code because the mechanism for enforcing the timeout is cancellation, i.e. we'd expect a `CancellationError` here.

Also this, this test _should_ be working at the moment because we enforce the timeout on the client side, it's just that we don't yet propagate it to the server.

Reply:
You're right. I'll get rid of this test for now: if we end up finding a way to implement timeouts in such a way that we don't expose the CancellationError, we can add this test back. Otherwise, we can't implement it and should remain gone. I'll create a follow-up PR to fix the propagation of CancellationError though, since this is not actually working (and the test failed/evidenced this)